src/util/logger.rs

```diff
@@ -97,3 +103,320 @@ fn test_get_level_by_string() {
+        $s.decorator.start_key()?;
+        write!($s.decorator, "{}", $k)?;
+        $s.decorator.start_separator()?;
+        write!($s.decorator, ":")?;
+        $s.decorator.start_whitespace()?;
+        write!($s.decorator, " ")?;
+        $s.decorator.start_value()?;
+        write!($s.decorator, "{}", $v)?;
+    };
+);
+
+#[cfg_attr(feature = "cargo-clippy", allow(write_literal))]
+impl<'a> slog::ser::Serializer for Serializer<'a> {
+    fn emit_none(&mut self, key: Key) -> slog::Result {
+        s!(self, key, "None");
+        Ok(())
+    }
+    fn emit_unit(&mut self, key: Key) -> slog::Result {
+        s!(self, key, "()");
+        Ok(())
+    }
+
+    fn emit_bool(&mut self, key: Key, val: bool) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+
+    fn emit_char(&mut self, key: Key, val: char) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+
+    fn emit_usize(&mut self, key: Key, val: usize) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_isize(&mut self, key: Key, val: isize) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+
+    fn emit_u8(&mut self, key: Key, val: u8) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_i8(&mut self, key: Key, val: i8) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_u16(&mut self, key: Key, val: u16) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_i16(&mut self, key: Key, val: i16) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_u32(&mut self, key: Key, val: u32) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_i32(&mut self, key: Key, val: i32) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_f32(&mut self, key: Key, val: f32) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_u64(&mut self, key: Key, val: u64) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_i64(&mut self, key: Key, val: i64) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_f64(&mut self, key: Key, val: f64) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_str(&mut self, key: Key, val: &str) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+    fn emit_arguments(&mut self, key: Key, val: &fmt::Arguments) -> slog::Result {
+        s!(self, key, val);
+        Ok(())
+    }
+}
+
+#[test]
+fn test_log_format() {
+    use chrono::{TimeZone, Utc};
+    use slog::Logger;
+    use slog_term::PlainSyncDecorator;
+    use std::cell::RefCell;
+    use std::io::Write;
+    use std::str::from_utf8;
+
+    // Due to the requirements of `Logger::root*` on a writer with a 'static lifetime
+    // we need to make a Thread Local,
+    // and implement a custom writer.
+    thread_local! {
```

Comment:
How about `lazy_static!`? I worry about context switch may fail the test.

Reply:
Context switch should not change thread local variables.

Reply:
What if context switch happens between L398 nad L409?

Reply:
I can't see the problem. Could you illustrate the detail of your concern?

Reply:
You are right, there's nothing to worry about!